File: app/src/main/java/com/pixels/blockies/app/game/figures/AbstractFigure.java
package com.pixels.blockies.app.game.figures;

public abstract class AbstractFigure implements Rotatable {
    private int currentRotation = 0;

    protected abstract int getFigureCount();

    protected int getCurrentRotation() {
        return currentRotation;
    }

    protected int getNextRotation() {
        if (currentRotation < getFigureCount() - 1) {
            return currentRotation+1;
        } else {
            return 0;
        }
    }

    public void rotate() {
        if (currentRotation < getFigureCount() - 1) {
            currentRotation++;
        } else {
            currentRotation = 0;
        }
    }

}

File: app/src/main/java/com/pixels/blockies/game/game/figures/AbstractFigure.kt
package com.pixels.blockies.game.game.figures

abstract class AbstractFigure : Rotatable {
    protected var currentRotation = 0
        private set
    protected abstract fun getFigureCount(): Int

    protected val nextRotation: Int
        get() = if (currentRotation < getFigureCount() - 1) {
            currentRotation + 1
        } else {
            0
        }

    override fun rotate() {
        if (currentRotation < getFigureCount() - 1) {
            currentRotation++
        } else {
            currentRotation = 0
        }
    }
}